facebook/react-native
# Implement onKeyPress Android

## Motivation

This implements onKeyPress for Android on TextInputs and addresses https://github.com/facebook/react-native/issues/1882. 
**N.B. that this PR has not yet addressed hardware keyboard inputs**, but doing will be fairly trivial. The main challenge was doing this for soft keyboard inputs. 

I've tried to match the style as much as I could. Will happily make any suggested edits be they architectural or stylistic design (edit: and of course implementation), but hopefully this is a good first pass :). 
I think important to test this on the most popular keyboard types; maybe different languages too.
I have not yet added tests to test implementation, but will be happy to do that also. 

## Test Plan
- Build & run RNTester project for Android and open TextInput. 
- Enter keys into 'event handling' TextInput. 
- Verify that keys you enter appear in onKeyPress below the text input
- Test with autocorrect off, on same input and validate that results are the same. 

Below is a gif of PR in action. 
![onkeypressandroid](https://user-images.githubusercontent.com/1807207/27512892-3f95c098-5949-11e7-9364-3ce9437f7bb9.gif)
